pkg/queryfrontend/roundtrip_test.go

```diff
@@ -280,6 +283,36 @@ func TestRoundTripSplitIntervalMiddleware(t *testing.T) {
 			splitInterval: 1 * time.Hour,
 			expected:      2,
 		},
+		{
+			name:                "split to 4 requests, due to min horizontal shards",
+			req:                 testRequest,
+			handlerFunc:         promqlResults,
+			codec:               queryRangeCodec,
+			splitInterval:       0,
+			querySplitThreshold: 30 * time.Minute,
+			maxSplitInterval:    4 * time.Hour,
+			minHorizontalShards: 4,
+			expected:            4,
+		},
+		{
+			name:                "split to 2 requests, due to maxSplitInterval",
+			req:                 testRequest,
+			handlerFunc:         promqlResults,
+			codec:               queryRangeCodec,
+			splitInterval:       0,
+			querySplitThreshold: 30 * time.Minute,
+			maxSplitInterval:    1 * time.Hour,
+			minHorizontalShards: 4,
+			expected:            2,
+		},
+		{
+			name:          "split to 2 requests, due to split interval",
+			req:           testRequest,
+			handlerFunc:   promqlResults,
+			codec:         queryRangeCodec,
+			splitInterval: 1 * time.Hour,
+			expected:      2,
```

Comment:
This test case is the same as the same at line 279?

Reply:
Good catch, thank you!